pyne/mcnp.py

```diff
@@ -1990,210 +1987,235 @@ def _read_meshtal_head(self, f):
-        The minimum and maximum bounds for energy bins
-    mesh :
-        An PyMOAB core instance tagged with all results and
-        relative errors
-    tag_names : iterable
-        Four strs that specify the tag names for the results, relative errors,
-        total results, and relative errors of the total results.
-
-    Notes
-    -----
-    All Mesh/StatMesh attributes are also present via a super() call to
-    StatMesh.__init__().
-
-    """
-
-    def __init__(self, f, tally_number, tag_names=None, mesh_has_mats=False):
-        """Create MeshTally object from a filestream open to the second
-        line of a mesh tally header (the neutron/photon line). MeshTally objects
-        should be instantiated only through the Meshtal class.
-
-        Parameters
-        ----------
-        f : filestream
-            Open to the neutron/photon line.
+        Parameters:
+        -----------
+        f : str or filestream
+            Filestream of the meshtal file.
         tally_number : int
             The MCNP fmesh4 tally number (e.g. 4, 14, 24).
+        particle : str
+            The particle type, 'neutron' or 'photon'.
         tag_names : iterable, optional
             Four strs that specify the tag names for the results, relative
             errors, total results and relative errors of the total results.
             This should come from the Meshtal.tags attribute dict.
         mesh_has_mats : bool
              If false, Meshtally objects will be created without PyNE material
              objects.
-        """
-
-        if not HAVE_PYMOAB:
-            raise RuntimeError("PyMOAB is not available, "
-                               "unable to create Meshtally Mesh.")
 
-        self.tally_number = tally_number
-        self._read_meshtally_head(f)
-        self._read_column_order(f)
+        Returns:
+        --------
+        m : MeshTally object
+            The MeshTally object created from MCNP mesh tally of tally_number.
+        """
 
+        m = MeshTally()
+        # assign tally_number
+        m.tally_number = tally_number
+        # assign particle and dose_response
+        m.particle, m.dose_response = self.read_meshtally_head(f)
+        # assign tag_names
         if tag_names is None:
-            self.tag_names = ("{0}_result".format(self.particle),
-                              "{0}_result_rel_error".format(self.particle),
-                              "{0}_result_total".format(self.particle),
-                              "{0}_result_total_rel_error".format(self.particle))
+            m.tag_names = ("{0}_result".format(m.particle),
+                              "{0}_result_rel_error".format(m.particle),
+                              "{0}_result_total".format(m.particle),
+                              "{0}_result_total_rel_error".format(m.particle))
         else:
-            self.tag_names = tag_names
-
-        self._create_mesh(f, mesh_has_mats)
+            m.tag_names = tag_names
```

Comment:
Should we do some type-checking to make sure we're getting an Iterable of 4 strs here?

Reply:
The type-checking for `tag_names` are added in https://github.com/pyne/pyne/pull/1199/commits/3d21bf4b7ce522fc31dd1f06d217edecb6bf55e9.